From 6146470cee9b6eb5252be4298a78b34150aed5fe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan-Simon=20M=C3=B6ller?= <dl9pf@gmx.de>
Date: Wed, 14 Nov 2012 10:52:18 +0100
Subject: [PATCH 07/51] Add noinline_for_stack to net/sctp/socket.c
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Author:  PaX Team <pageexec at freemail.hu>
ML-Post: http://lists.cs.uiuc.edu/pipermail/llvm-commits/Week-of-Mon-20120507/142707.html
URL:     http://llvm.linuxfoundation.org

Merge:   Jan-Simon Möller <dl9pf at gmx.de>

Description:
Add noinline_for_stack to net/sctp/socket.c
---
 net/sctp/socket.c |   24 ++++++++++++------------
 1 Datei geändert, 12 Zeilen hinzugefügt(+), 12 Zeilen entfernt(-)

diff --git a/net/sctp/socket.c b/net/sctp/socket.c
index a60d1f8..f853944 100644
--- a/net/sctp/socket.c
+++ b/net/sctp/socket.c
@@ -2458,7 +2458,7 @@ static int sctp_apply_peer_addr_params(struct sctp_paddrparams *params,
 	return 0;
 }
 
-static int sctp_setsockopt_peer_addr_params(struct sock *sk,
+static noinline_for_stack int sctp_setsockopt_peer_addr_params(struct sock *sk,
 					    char __user *optval,
 					    unsigned int optlen)
 {
@@ -2723,7 +2723,7 @@ static int sctp_setsockopt_initmsg(struct sock *sk, char __user *optval, unsigne
  *   sinfo_timetolive.  The user must provide the sinfo_assoc_id field in
  *   to this call if the caller is using the UDP model.
  */
-static int sctp_setsockopt_default_send_param(struct sock *sk,
+static noinline_for_stack int sctp_setsockopt_default_send_param(struct sock *sk,
 					      char __user *optval,
 					      unsigned int optlen)
 {
@@ -2763,7 +2763,7 @@ static int sctp_setsockopt_default_send_param(struct sock *sk,
  * the association primary.  The enclosed address must be one of the
  * association peer's addresses.
  */
-static int sctp_setsockopt_primary_addr(struct sock *sk, char __user *optval,
+static noinline_for_stack int sctp_setsockopt_primary_addr(struct sock *sk, char __user *optval,
 					unsigned int optlen)
 {
 	struct sctp_prim prim;
@@ -3039,7 +3039,7 @@ static int sctp_setsockopt_maxseg(struct sock *sk, char __user *optval, unsigned
  *   locally bound addresses. The following structure is used to make a
  *   set primary request:
  */
-static int sctp_setsockopt_peer_primary_addr(struct sock *sk, char __user *optval,
+static noinline_for_stack int sctp_setsockopt_peer_primary_addr(struct sock *sk, char __user *optval,
 					     unsigned int optlen)
 {
 	struct net *net = sock_net(sk);
@@ -4051,7 +4051,7 @@ SCTP_STATIC void sctp_shutdown(struct sock *sk, int how)
  * number of unacked data chunks, and number of data chunks pending
  * receipt.  This information is read-only.
  */
-static int sctp_getsockopt_sctp_status(struct sock *sk, int len,
+static noinline_for_stack int sctp_getsockopt_sctp_status(struct sock *sk, int len,
 				       char __user *optval,
 				       int __user *optlen)
 {
@@ -4131,7 +4131,7 @@ out:
  * window, and retransmission timer values.  This information is
  * read-only.
  */
-static int sctp_getsockopt_peer_addr_info(struct sock *sk, int len,
+static noinline_for_stack int sctp_getsockopt_peer_addr_info(struct sock *sk, int len,
 					  char __user *optval,
 					  int __user *optlen)
 {
@@ -4433,7 +4433,7 @@ out:
  *                     SPP_SACKDELAY_ENABLE, setting both will have undefined
  *                     results.
  */
-static int sctp_getsockopt_peer_addr_params(struct sock *sk, int len,
+static noinline_for_stack int sctp_getsockopt_peer_addr_params(struct sock *sk, int len,
 					    char __user *optval, int __user *optlen)
 {
 	struct sctp_paddrparams  params;
@@ -4626,7 +4626,7 @@ static int sctp_getsockopt_initmsg(struct sock *sk, int len, char __user *optval
 }
 
 
-static int sctp_getsockopt_peer_addrs(struct sock *sk, int len,
+static noinline_for_stack int sctp_getsockopt_peer_addrs(struct sock *sk, int len,
 				      char __user *optval, int __user *optlen)
 {
 	struct sctp_association *asoc;
@@ -4722,7 +4722,7 @@ static int sctp_copy_laddrs(struct sock *sk, __u16 port, void *to,
 }
 
 
-static int sctp_getsockopt_local_addrs(struct sock *sk, int len,
+static noinline_for_stack int sctp_getsockopt_local_addrs(struct sock *sk, int len,
 				       char __user *optval, int __user *optlen)
 {
 	struct sctp_bind_addr *bp;
@@ -4827,7 +4827,7 @@ out:
  * the association primary.  The enclosed address must be one of the
  * association peer's addresses.
  */
-static int sctp_getsockopt_primary_addr(struct sock *sk, int len,
+static noinline_for_stack int sctp_getsockopt_primary_addr(struct sock *sk, int len,
 					char __user *optval, int __user *optlen)
 {
 	struct sctp_prim prim;
@@ -4908,7 +4908,7 @@ static int sctp_getsockopt_adaptation_layer(struct sock *sk, int len,
  *
  *   For getsockopt, it get the default sctp_sndrcvinfo structure.
  */
-static int sctp_getsockopt_default_send_param(struct sock *sk,
+static noinline_for_stack int sctp_getsockopt_default_send_param(struct sock *sk,
 					int len, char __user *optval,
 					int __user *optlen)
 {
@@ -5042,7 +5042,7 @@ static int sctp_getsockopt_rtoinfo(struct sock *sk, int len,
  * See [SCTP] for more information.
  *
  */
-static int sctp_getsockopt_associnfo(struct sock *sk, int len,
+static noinline_for_stack int sctp_getsockopt_associnfo(struct sock *sk, int len,
 				     char __user *optval,
 				     int __user *optlen)
 {
-- 
1.7.10.4

